src/platform/packages/shared/kbn-esql-ast/src/commands_registry/commands/rerank/validate.ts

```diff
@@ -20,35 +32,72 @@ export const validate = (
 ): ESQLMessage[] => {
   const messages: ESQLMessage[] = [];
 
-  // Run standard argument validation
-  messages.push(...validateCommandArguments(command, ast, context, callbacks));
+  const { query, targetField, location } = command as ESQLAstRerankCommand;
+
+  // Sets the target field so the column is recognized after the command is applied
+  const targetName = targetField?.name || 'rerank';
+
+  context?.userDefinedColumns.set(targetName, [
+    {
+      name: targetName,
+      location: targetField?.location || location,
+      type: 'keyword',
+    },
+  ]);
 
-  // Cast to RERANK command for type-specific validation
-  const rerankCommand = command as ESQLAstRerankCommand;
+  const rerankExpressionType = getExpressionType(
+    query,
+    context?.fields,
+    context?.userDefinedColumns
+  );
 
-  // Validate that query text is provided
-  if (!rerankCommand.query) {
+  // check for supported query types
+  if (!supportedQueryTypes.includes(rerankExpressionType)) {
     messages.push({
-      location: command.location,
-      text: i18n.translate('kbn-esql-ast.esql.validation.rerankMissingQuery', {
-        defaultMessage: '[RERANK] Query text is required.',
+      location: 'location' in query ? query?.location : location,
+      text: i18n.translate('kbn-esql-ast.esql.validation.rerankUnsupportedFieldType', {
+        defaultMessage: '[RERANK] query must be of type [text] but is [{rerankExpressionType}]',
```

Comment:
```suggestion
        defaultMessage: 'RERANK query must be of type text. Found {rerankExpressionType}',
```

(the style of our error messages changed a bit in https://github.com/elastic/kibana/pull/233093)